pkg/controller/controllerutil/controllerutil.go

```diff
@@ -377,10 +377,3 @@ func ContainsFinalizer(o Object, finalizer string) bool {
 	}
 	return false
 }
-
-// Object allows functions to work indistinctly with any resource that
-// implements both Object interfaces.
-type Object interface {
```

Comment:
Maybe throw in a `type Object = client.Object` to ease the transition?

Reply:
Maybe? It could be a good opportunity to finally move this interface somewhere else :)

Reply:
It's a breaking change without that, seems unnecessary.

Reply:
I tend to agree, because its not obvious without reading release notes where this went. What about keeping the alias for a minor release add adding a `Deprecated: Use from client package instead` comment?

Reply:
Yeah, there are tons of breaking changes in this release. I'm happy to hear more opinions from others as well, I can definitely add the alias, although the function signature will change regardless, so 🤷‍♂️

Reply:
Well the definition of this interface stays the same. This is mostly relevant for ppl who use this interface in their code, for example as the type for a function argument.

Reply:
Done, ptal

Reply:
yeah, that's generally a good idea.

Can mark the alias as deprecated though.